static/js/GuideBox.jsx

```diff
@@ -0,0 +1,197 @@
+import TextRange from "./TextRange";
+import {AiInfoTooltip} from './Misc';
+
+// Constants for different prompt states
+const QUESTIONS = "questions";
+const SUMMARIES = "summaries";
+const COMMENTARIES = "commentaries";
+
+/**
+ * Renders the question prompts in a box.
+ * @param {Array} props.prompt - The array of question prompts.
+ * @param {Function} props.onClick - The function to handle click events on prompts.
+ * @returns {JSX.Element} The QuestionBox component.
+ */
+const QuestionBox = ({ prompt, onClick }) => {
+
+  const questionHelperText = (p) => {
+    return `${p.commentaries.length} ${p.commentaries.length > 0 ? 'Answers' : 'Answer'}`;
+  }
+
+  return (
+    <div>
+      <div className="guideTitle">
+        Key Questions
+      </div>
+      {prompt.map((p, i) => {
+        return (
+          <div key={i} className="guidePromptBox" onClick={() => onClick(p, i)}>
+            <p>{p.question}</p>
+            <span>{questionHelperText(p)}</span>
+          </div>
+        )
+      })}
+    </div>
+  );
+}
+
+/**
+ * Renders the summary prompts in a box.
+ * @param {Object} props.prompt - The summary prompt object.
+ * @param {Function} props.onClick - The function to handle click events on prompts.
+ * @returns {JSX.Element} The SummaryBox component.
+ */
+const SummaryBox = ({ prompt, onClick }) => {
+  const title = (commentaryRef) => {
+    const index = Sefaria.parseRef(commentaryRef)?.index;
+    return index ? index.replace(/ on Pesach Haggadah$/, "") : commentaryRef;
+  }
+
+  return (
+    <div>
+      <div className="guideTitle">
+        {prompt.question}
+      </div>
+      {prompt.commentaries.map((p, i) => {
+        return (
+          <div key={i} className="guidePromptBox" onClick={() => onClick(p.commentaryRef, i)}>
+            <p>{p.summaryText}</p>
+            <span>{title(p.commentaryRef)}</span>
+          </div>
+        )
+      })}
+    </div>
+  );
+}
+
+
+class GuideBox extends Component {
+  constructor(props) {
+    super(props);
+    const guides = Sefaria.guidesByRef(props.sref);
+    const guide = guides[0];
+    this.state = {
+      guideLanguage: Sefaria.interfaceLang,
+      guide: guide,
+      livePrompt: guide.questions,
+      questionPosition: 0,
+      promptState: QUESTIONS,
+      commentaryRef: ""
+    };
+    this.stateHistory = [];
+  }
+
+  componentDidUpdate(prevProps) {
+    if (this.props.sref !== prevProps.sref) {
+      const guides = Sefaria.guidesByRef(this.props.sref);
+      if (guides.length) {
+        const guide = guides[0];
+        this.setState({ guide: guide, livePrompt: guide.questions, questionPosition: 0, promptState: QUESTIONS });
+        this.resetHistory();
+      }
+    }
+  }
+
+  /**
+   * Overrides the setState method to add the current state to the history before updating it.
+   * @param {Object} newState - The new state object.
+   */
+  setState(newState) {
+    this.stateHistory.push(this.state);
+    super.setState(newState);
```

Comment:
I override setState to keep a history of the state and send `popState` to the parent component so that it can control the level of the child component generically.